windows/src/developer/TIKE/web/UmodWebHttpServer.pas

```diff
@@ -437,92 +441,49 @@ procedure TmodWebHttpServer.httpCommandGet(AContext: TIdContext;
       end;
 
       value := FKeyboards[filename];
-      if value.JSONFilename = '' then
-      begin
-        Respond404;
-        Exit;
-      end;
-
-      with TStringStream.Create('', TEncoding.UTF8) do
-      try
-        LoadFromFile(value.JSONFilename);
-        FJSONText := DataString;
-      finally
-        Free;
-      end;
 
-      Offset := 0;
+      // Build the JSON response
 
+      json := TJSONObject.Create;
       try
-        json := ParseJSONValue(TrimRight(FJSONText), Offset) as TJSONObject;
-        if not Assigned(json) then
-        begin
-          Respond404;
-          Exit;
-        end;
-      except
-        on E:EJSONException do
-        begin
-          Respond404;
-          Exit;
-        end;
-      end;
+        jsonOptions := TJSONObject.Create;
+        jsonKeyboard := TJSONObject.Create;
 
-      try
-        try
-          jsonOptionsPair := json.Get('options');
-          jsonKeyboardPair := json.Get('keyboard');
-          if (jsonOptionsPair = nil) or (jsonKeyboardPair = nil) then
-          begin
-            Respond404;
-            Exit;
-          end;
+        json.AddPair('options', jsonOptions)
+            .AddPair('keyboard', jsonKeyboard);
 
-          if not (jsonOptionsPair.JsonValue is TJSONObject) or
-             not (jsonKeyboardPair.JsonValue is TJSONObject) then
-          begin
-            Respond404;
-            Exit;
-          end;
-
-          jsonOptions := jsonOptionsPair.JsonValue as TJSONObject;
-          jsonKeyboard := jsonKeyboardPair.JsonValue as TJSONObject;
+        FKeyboardBaseUri := Format('http://%s/keyboard/',[ARequestInfo.Host]);
+        FFontBaseUri := Format('http://%s/font/',[ARequestInfo.Host]);
 
-          FKeyboardBaseUri := Format('http://%s/keyboard/',[ARequestInfo.Host]);
-          FFontBaseUri := Format('http://%s/font/',[ARequestInfo.Host]);
+        jsonOptions.AddPair('device', 'any');
+        jsonOptions.AddPair('keyboardBaseUri', FKeyboardBaseUri);
+        jsonOptions.AddPair('fontBaseUri', FFontBaseUri);
 
-          jsonOptions.RemovePair('keyboardBaseUri').Free;
-          jsonOptions.AddPair('keyboardBaseUri', FKeyboardBaseUri);
+        jsonFont := TJSONObject.Create;
+        jsonFont.AddPair('family', value.FontName[kfontChar]);   // I4409
+        jsonFont.AddPair('filename', value.ID+'-'+value.Version+'-'+TestFontName[kfontChar]+'.ttf');   // I4409
+        jsonKeyboard.AddPair('font', jsonFont);
 
-          jsonOptions.RemovePair('fontBaseUri').Free;
-          jsonOptions.AddPair('fontBaseUri', FFontBaseUri);
+        jsonOskFont := TJSONObject.Create;
+        jsonOskFont.AddPair('family', value.FontName[kfontOSK]);   // I4409
+        jsonOskFont.AddPair('filename', value.ID+'-'+value.Version+'-'+TestFontName[kfontOSK]+'.ttf');   // I4409
```

Comment:
Is `filename` correct? When I've tested legacy distribution, I've been using this JSON
```
    "font": {
      "family":"Khmer Mondulkiri",
      "source":"Mondulkiri-R.ttf"
    },
    "oskFont": {
      "family":"Khmer Mondulkiri",
      "source":"Mondulkiri-R.ttf"
    },
```

Reply:
Indeed you are right. `filename` was an old, wrong way of doing it.